libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/RouteComparator.java

```diff
@@ -0,0 +1,73 @@
+package com.mapbox.services.android.navigation.ui.v5;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse;
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.api.directions.v5.models.RouteLeg;
+
+import java.util.List;
+
+class RouteComparator {
+
+  private static final int FIRST_ROUTE = 0;
+  private static final int ONE_ROUTE = 1;
+  private final NavigationViewRouter navigationViewRouter;
+
+  RouteComparator(NavigationViewRouter navigationViewRouter) {
+    this.navigationViewRouter = navigationViewRouter;
+  }
+
+  void compare(@NonNull DirectionsResponse response, @Nullable DirectionsRoute chosenRoute) {
+    if (isValidRoute(response)) {
+      List<DirectionsRoute> routes = response.routes();
+      DirectionsRoute bestRoute = routes.get(FIRST_ROUTE);
+      if (isNavigationRunning(chosenRoute)) {
+        bestRoute = obtainMostSimilarRoute(routes, bestRoute, chosenRoute);
+      }
+      navigationViewRouter.updateCurrentRoute(bestRoute);
+    }
+  }
+
+  private DirectionsRoute obtainMostSimilarRoute(List<DirectionsRoute> routes, DirectionsRoute currentBestRoute,
```

Comment:
This feels redundant with `findMostSimilarRoute`, if you don't pass in the first route you could just get rid of this method and call the other one.